### PR TITLE
feat: migrate to 3.0

### PR DESCRIPTION
**Pull request by winstonww**
Diff: https://github.com/jina-ai/executors/pull/323

A workaround is introduced to make matches' ids unique, but this should be removed once issue is addressed.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+repos:
+  - repo: https://github.com/psf/black
+    rev: 21.7b0
+    hooks:
+      - id: black
+        types: [python]
+        args:
+        - -S
+        #   - repo: https://github.com/pycqa/flake8
```
Comment (winstonww) [inline .pre-commit-config.yaml:9]: this is commented out now to generate diff but will enable this before merging

File: manifest.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+manifest_version: 1
+name: FaissPostgresIndexer
+description: A compound FAISS- and PostgreSQL-based query indexer
+url: https://github.com/jina-ai/executors/tree/main/jinahub/indexers/compound/FaissPostgresIndexer
```
Comment (nan-wang) [inline manifest.yml:4]: please update this url

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
+FROM jinaai/jina:2-py37-perf as base
```
Comment (numb3r3) [inline Dockerfile:1]: use jina 3 base image.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+# install and upgrade pip
+RUN apt-get update && apt-get install -y git
```
Comment (numb3r3) [inline Dockerfile:4]: `git` can be removed here.

File: config.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+jtype: FaissPostgresIndexer
+with:
+  {}
+metas:
+  py_modules:
+    - executor/__init__.py
+  workspace: workspace/
+  name: faiss_postgres_searcher
+runtime_args:
+  name: faiss_postgres_searcher
```
Comment (numb3r3) [inline config.yml:10]: check the above args

File: .pre-commit-config.yaml (hunk 1 above)
Comment (numb3r3) [reply .pre-commit-config.yaml:9]: you can borrow some code from docarray, finetuner etc.,

**Review by numb3r3 — CHANGES_REQUESTED** [review]

**Comment by nan-wang** [comment]
Close the PR because there is no updates for quite a while.

**Pull request closed without merging**